### ISSUE TITLE
Create/use lutaml-path (something like object path, VPath, XPath but for lutaml-model)

### ISSUE THREAD

**Issue by HassanAkbar** [lutaml/lutaml-model#173]

**Comment by HassanAkbar** [comment]
@ronaldtse From what I understand we need to support something like what xpath does i.e `lutaml_model_instance.model_path(<expression>)`.

Is this correct?

**Comment by ronaldtse** [comment]
Yes, that's correct. What we are doing in root_mappings / child_mappings is like that:

```
    map_root to: :units, root_mappings: { id: :key, name: %w[root_units enumerated_root_units] }
```

i.e. `root_units enumerated_root_units` => in YAML `input.root_units.enumerated_root_units`.

We need to think about it.

**Comment by ronaldtse** [comment]
We have a gem for [lutaml-path](https://github.com/lutaml/lutaml-path).

We need a specification for lutaml-path and to extend it.
